src/app/modules/portal-shared/components/confirm-dialog/confirm-dialog.component.ts

```diff
@@ -0,0 +1,48 @@
+import { MatDialogRef, MAT_DIALOG_DATA } from '@angular/material/dialog';
+import { Component, OnInit, Inject } from '@angular/core';
+
+@Component({
+  selector: 'app-confirm-dialog',
+  templateUrl: './confirm-dialog.component.html',
+  styleUrls: ['./confirm-dialog.component.scss']
+})
+export class ConfirmDialogComponent implements OnInit {
+  title: string;
+  message: string;
+  cancelButtonText: string;
+  confirmButtonText: string;
+  constructor(public dialogRef: MatDialogRef<ConfirmDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: ConfirmDialogModel) {
+    // Update view with given values
+    this.title = data.title;
+    this.message = data.message;
+    this.cancelButtonText = data['cancelButtonText'];
+    this.confirmButtonText = data['confirmButtonText'];
+  }
+
+  ngOnInit() {
+  }
+
+  onConfirm(): void {
+    // Close the dialog, return true
+    this.dialogRef.close(true);
+  }
+
+
+  onDismiss(): void {
+    // Close the dialog, return false
+    this.dialogRef.close(false);
+  }
+}
+
+/**
+ * Class to represent confirm dialog model.
+ *
+ * It has been kept here to keep it as part of shared component.
+ */
+export class ConfirmDialogModel {
```

Comment:
What is the use of this?

Reply:
We are 'using to send message, title, confirm and cancel buttons, same as like interface concept purpose we are using here